### PR TITLE
workload cmd: set discoveryaddress via ISTIOD_ADDRESS

### PR DESCRIPTION
**Pull request by stevenctl**
safer, less robust, alternative to #35290

### CODE REVIEW

File: pkg/test/framework/components/namespace/kube.go [hunk 1]
```diff
@@ -76,6 +79,34 @@ func (n *kubeNamespace) Prefix() string {
...
+	}
+	if err := errG.Wait().ErrorOrNil(); err != nil {
+		return nil, err
+	}
+	for i, clusterLabels := range perCluster {
+		if i == 0 {
+			continue
+		}
+		if diff := cmp.Diff(perCluster[0], clusterLabels); diff != "" {
+			log.Warnf("namespace labels are different accross clusters:\n%s", diff)
```
Comment (stevenctl) [inline pkg/test/framework/components/namespace/kube.go:104]:
```suggestion
			log.Warnf("namespace labels are different across clusters:\n%s", diff)
```

File: istioctl/cmd/workload.go [hunk 2]
```diff
@@ -378,9 +381,12 @@ func createSidecarEnv(internalIP string, externalIP, dir string) error {
 	return os.WriteFile(filepath.Join(dir, "sidecar.env"), []byte(mapToString(sidecarEnv)), filePerms)
 }
 
-func generateSidecarEnvAsMap(internalIP string, externalIP string) map[string]string {
+func generateSidecarEnvAsMap(internalIP string, externalIP string, rev string) map[string]string {
 	sidecarEnv := make(map[string]string)
 
+	if addr := revisionedIstiodAddress(rev); addr != "" {
+		sidecarEnv["PILOT_ADDRESS"] = addr
```
Comment (howardjohn) [inline istioctl/cmd/workload.go:388]: Why do we need this if we have the EXPLICITLY_SET_PILOT_ADDRESS thing?

**Review by howardjohn — COMMENTED** [review]
Did we ever figure out why the test wasn't effective?

File: istioctl/cmd/workload.go (hunk 2 above)
Comment (stevenctl) [reply istioctl/cmd/workload.go:388]: We don't -- I didn't revert it correctly.

**Comment by stevenctl** [comment]
> Did we ever figure out why the test wasn't effective?

One issue is that we didn't actually pass `--revision` to `istioctl`, so we weren't even setting the value. 
This means the discovery address was defaulting to `istiod.istio-system...` which I guess still exists in the revisions test.

File: tests/integration/pilot/revisions/revisions_test.go [hunk 3]
```diff
@@ -95,9 +95,10 @@ func TestMultiRevision(t *testing.T) {
 				}).
 				// tests bootstrap
 				WithConfig(echo.Config{
-					Service:   "vm",
-					Namespace: canary,
-					Ports:     []echo.Port{},
+					Service:    "vm",
+					Namespace:  canary,
+					DeployAsVM: true,
```
Comment (stevenctl) [inline tests/integration/pilot/revisions/revisions_test.go:100]: This is also why the test didn't give coverage...

**Pull request closed without merging**